message_post_model module can now dynamically activate the track fields changes in the models message logs (even exclude some fields if wanted)

josemoralesp opened the pull request:
To ease the way that "message_post_model"  works and its integration with models with a complex workflow, we made some improvements over this module and the way that this is tested.

The first change was the way that you specify which models you want to track their changes. Previously you had to create an inheritance over the removed object show.all.message. Now you can do this without change any code, you can specify directly the model that you want to track using a new server action that you can use in ir_model views(tree and form). It is necessary that the model has at least an inheritance over mail.thread.

Now you can specify if you want to exclude some fields of the tracking, to avoid generate a log message if these are changed.

The tracking can be added and removed of the models, with this change we can avoid stop a workflow if this module generates an unexpected error.

Commit:
[IMP] The tracked of all fields in a object is now dynamically

Delete no used anymore model_message_post_show_all model

zaoral commented on message_post_model/README.rst:
```diff
@@ -1,16 +1,57 @@
+
+In the ir.model object we added three new fields:
+
+    - .. image:: https://drive.google.com/uc?export=view&id=0B2kzKLGF6ZvLRFYwcTBYT2ZUQ1E
```
``` diff
-    - .. image:: https://drive.google.com/uc?export=view&id=0B2kzKLGF6ZvLRFYwcTBYT2ZUQ1E
+  .. image:: https://drive.google.com/uc?export=view&id=0B2kzKLGF6ZvLRFYwcTBYT2ZUQ1E
```

I think for aesthetic sake that this image should be out of the unordered list.
This is is how it looks right now 

![image](https://cloud.githubusercontent.com/assets/7593953/19330508/de4ca406-90ac-11e6-93d6-aba8d2b1bc0d.png)

zaoral commented on message_post_model/__openerp__.py:
```diff
@@ -34,7 +34,8 @@
     ],
     "demo": [],
     "data": [
-        "security/ir.model.access.csv"
```
With this change runbot build now give us the next warning 

``` bash
2016-10-09 16:56:16,514 254 WARNING openerp_test openerp.modules.loading: The model message.post.show.all has no access rules, consider adding one. E.g. access_message_post_show_all,access_message_post_show_all,model_message_post_show_all,,1,0,0,0
```

josemoralesp replied:
The same answer:
https://github.com/Vauxoo/addons-vauxoo/pull/1075#discussion_r83271412

zaoral replied:
ready

zaoral commented on message_post_model/i18n/es.po:
```diff
@@ -28,125 +28,100 @@ msgstr ""
+#: code:addons/message_post_model/model/message_post.py:376
+#, python-format
+msgid "Warning!"
+msgstr "Warning!"
```
missing translation

zaoral commented on message_post_model/i18n/es_MX.po:
```diff
@@ -6,19 +6,19 @@ msgid ""
 msgstr ""
```
@josemoralesp 
I think that we do not need to re-define all the Spanish translations in es.po at es_MX.po
I think that we can manage only a es_MX.po file with the head of the file. Something like this

```
# Translation of Odoo Server.
# This file contains the translation of the following modules:
#   * message_post_model
#
msgid ""
msgstr ""

"Project-Id-Version: Odoo Server 8.0\n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: 2016-10-07 18:02+0000\n"
"PO-Revision-Date: 2016-10-07 13:08-0500\n"
"Last-Translator: <>\n"
"Language-Team: \n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Plural-Forms: \n"
"Language: es\n"
"X-Generator: Poedit 1.8.7.1\n"
```

Will be easy to maintain the translations and es_MX.po will reuse the Spanish translations.

zaoral replied:
ready

zaoral commented on message_post_model/i18n/es_VE.po:
```diff
@@ -6,19 +6,19 @@ msgid ""
 msgstr ""
```
Same case here, only need to have es_VE.po with the head of the po file.

zaoral replied:
ready

zaoral commented on message_post_model/i18n/message_post_model.pot:
```diff
@@ -1,147 +1,127 @@
 #, python-format
 msgid "Added"
-msgstr ""
+msgstr "Agregado"
```
I think there is some confusion when updating this file.
Should be pot file template, the should not be translation inside.

zaoral commented on message_post_model/static/description/index.html:
```diff
@@ -1,76 +1,67 @@
+            </li>
+            <li>
+                <dl>
+                    <dt>To remove it:</dt>
```
I not sure why the rst syntax set `To remove it` as a internal list, I think should be in the same level list of `To add the track`

josemoralesp replied:
It was created using rst2html

zaoral commented on message_post_model/static/description/index.html:
```diff
@@ -1,76 +1,67 @@
+            <img src="https://www.vauxoo.com/logo.png" alt="Vauxoo" />
+        </a>
+        <p>This module is maintained by Vauxoo.</p>
+        <p>A latinamerican company that provides training, coaching, development and implementation of enterprise management sytems and bases its entire operation strategy in the use of Open Source Software and its main product is odoo.</p>
```
also if you use odoo-description (gist-vauxoo/description2html script) you could have a pretty format printed html file.

zaoral commented on message_post_model/model/message_post.py:
```diff
@@ -22,273 +22,390 @@
+
+class MessagePostShowAll(models.Model):
+
+    # TODO: DELETE THIS CLASS
```
Why do not delete this class in the current PR?

josemoralesp replied:
I don't delete the class, because there are more module with an inheritance over him.

I was testing the concept only with the yoytec module, then this class must be removed and all modules must be changed

zaoral replied:
ready

zaoral commented on message_post_model/model/message_post.py:
```diff
@@ -22,273 +22,390 @@
 
 
-class MessagePostShowAll(models.Model):
+# pylint: disable=W0622
```
I think that this pylint disable is not necessary anymore

``` diff
+# pylint: disable=W0622
```

zaoral replied:
ready

zaoral reviewed (changes requested):
Need to make some changes, functional review +1 however we had one problem, you can see the functional review [here](https://youtu.be/-IaKbQs5Zoo)

@zaoral Fixed the error shown in the video. I will create another runbot

zaoral commented:
Hi @josemoralesp 
What is missing to merge this PR?

josemoralesp commented:
Hello @zaoral

For merge this we need to prepare a serie of PR with all branches with dependencies with this module.

Best Regards.

zaoral commented:
Hi @josemoralesp 
I understand, please can you help us with this task?

zaoral committed:
[FIX] update commission_payment to match with new version of
message_post_model module

zaoral committed:
[ADD] message_post_model: Add es_PA.po translation

zaoral committed:
[pylint/flake] Fix multiple errors and warnings in commission_payment module

- W1619 (old-division)
- W503 line break before binary operator
- C0411 (wrong-import-order) standard import

zaoral commented:
Hi @moylop260 
Please could you help me with this merge.
After merge this one we need to merge the next one to update the modules that depends on message_post_model.
- https://github.com/Vauxoo/yoytec/pull/2043
- https://github.com/Vauxoo/purchase/issues/13
